open-metadata-implementation/governance-servers/open-lineage-services/open-lineage-services-server/src/main/java/org/odpi/openmetadata/governanceservers/openlineage/scheduler/BufferGraphJob.java

```diff
@@ -3,34 +3,32 @@
 package org.odpi.openmetadata.governanceservers.openlineage.scheduler;
 
 import org.odpi.openmetadata.governanceservers.openlineage.buffergraph.BufferGraph;
-import org.quartz.Job;
-import org.quartz.JobDataMap;
-import org.quartz.JobExecutionContext;
-import org.quartz.JobKey;
+import org.quartz.*;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import java.time.LocalDateTime;
 
+@DisallowConcurrentExecution
 public class BufferGraphJob implements Job {
 
     private static final Logger log = LoggerFactory.getLogger(BufferGraphJob.class);
 
     @Override
     public void execute(JobExecutionContext context) {
         LocalDateTime localTime = LocalDateTime.now();
-        log.info("Run QuartzJob at " + localTime);
+        System.out.println("Run QuartzJob at " + localTime);
+
         JobKey key = context.getJobDetail().getKey();
+
         JobDataMap dataMap = context.getJobDetail().getJobDataMap();
+
         BufferGraph bufferGraph = (BufferGraph) dataMap.get("openLineageGraphStore");
+
         performTask(bufferGraph);
     }
 
-    private void performTask(BufferGraph bufferGraph) {
-        try {
-            bufferGraph.schedulerTask();  //TODO  Throws NPE
-        } catch (Exception e) {
-            log.error("Buffergraph job Scheduler error");
-        }
+    private void performTask(BufferGraph bufferGraph){
+        bufferGraph.schedulerTask();
```

Comment:
This task can throw an NPE if there is a problem with the buffer graph connection, should be catched.